cmake/ExternalProjects.cmake

```diff
@@ -62,7 +66,9 @@ if(FAABRIC_BUILD_TESTS)
         GIT_TAG "v2.13.2"
         CMAKE_ARGS "-DCATCH_INSTALL_DOCS=OFF \
             -DCATCH_INSTALL_EXTRAS=OFF"
-        CMAKE_CACHE_ARGS "-DCMAKE_INSTALL_PREFIX:STRING=${THIRD_PARTY_INSTALL_DIR}"
+        CMAKE_CACHE_ARGS "-DCMAKE_INSTALL_PREFIX:STRING=${CMAKE_INSTALL_PREFIX}"
     )
+    ExternalProject_Get_Property(catch_ext SOURCE_DIR)
+    include_directories(${CMAKE_INSTALL_PREFIX}/include/catch2)
```

Comment:
Catch2 requires different `#includes` depending on how it's installed. Doing it like this or from source means we have to write `#include <catch.hpp>` (as they do in their docs), rather than `#include <catch2/catch.hpp>`. This means there are hundreds of changes in the tests below.